src/main/java/io/supertokens/config/CoreConfig.java

```diff
@@ -718,14 +782,70 @@ public void createLoggingFile(Main main) throws IOException {
+        // set these dynamically.
         for (Field field : CoreConfig.class.getDeclaredFields()) {
             if (field.isAnnotationPresent(ConfigYamlOnly.class)) {
                 if (config.has(field.getName())) {
-                    throw new InvalidConfigException(field.getName() + " can only be set via the core's base config setting");
+                    throw new InvalidConfigException(
+                            field.getName() + " can only be set via the core's base config setting");
+                }
+            }
+        }
+    }
+
+    public static ArrayList<ConfigFieldInfo> getConfigFieldsInfo(Main main) {
+        ArrayList<ConfigFieldInfo> result = new ArrayList<ConfigFieldInfo>();
+
+        for (String fieldId : CoreConfig.getValidFields()) {
+            try {
+                Field field = CoreConfig.class.getDeclaredField(fieldId);
+                // If fieldId exists in PROTECTED_CONFIGS or is not annotated with JsonProperty
+                // or is annotated with ConfigYamlOnly, then skip
+                if (Arrays.asList(PROTECTED_CONFIGS).contains(fieldId)
+                        || !field.isAnnotationPresent(JsonProperty.class)
+                        || field.isAnnotationPresent(ConfigYamlOnly.class)
+                        || fieldId == "core_config_version") {
+                    continue;
+                }
+
+                String name = field.getName();
+                String description = field.isAnnotationPresent(ConfigDescription.class)
+                        ? field.getAnnotation(ConfigDescription.class).value()
+                        : "";
+                boolean isDifferentAcrossTenants = !field.isAnnotationPresent(NotConflictingInApp.class);
+
+                String type = "string";
+
+                Class<?> fieldType = field.getType();
+
+                if (fieldType == java.lang.String.class) {
+                    type = "string";
+                } else if (fieldType == boolean.class) {
+                    type = "boolean";
+                } else if (fieldType == byte.class || fieldType == short.class || fieldType == int.class
+                        || fieldType == long.class || fieldType == float.class || fieldType == double.class) {
+                    type = "number";
                 }
+
+                String[] options = null;
+
+                if (field.isAnnotationPresent(EnumProperty.class)) {
+                    type = "enum";
+                    options = field.getAnnotation(EnumProperty.class).value();
+                }
+                result.add(new ConfigFieldInfo(name, description, isDifferentAcrossTenants, type, options));
+
+            } catch (NoSuchFieldException e) {
+                continue;
             }
         }
+
+        ArrayList<ConfigFieldInfo> storageFields = StorageLayer.getBaseStorage(main).getConfigFieldsInfo();
+
+        result.addAll(storageFields);
+
+        return result;
```

Comment:
Move this to the API layer itself and remove from here.

Reply:
Moved it to API layer